src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs

```diff
@@ -33,7 +33,11 @@ public async Task AddServiceBusTopicMessagePump_WithSubscriptionNameIndirectSecr
             // Act
             ServiceBusMessageHandlerCollection result = 
                 services.AddServiceBusTopicMessagePump(
-                    "topic name", "subscription name", "secret name", options => options.AutoComplete = true);
+                    "topic name", "subscription name", "secret name", options =>
+                    {
+                        options.AutoComplete = true;
+                        options.TopicSubscription = TopicSubscription.CreateOnStart | TopicSubscription.DeleteOnStop;
```

Comment:
I'm actually wondering if it is necessary to have these 2 possibilities separated.
I think the pump should only be able to remove the subscription if the subscription has been auto-created.
In other words, I think it is useless to be able to only enable the `DeleteOnStop` option.  
I think we should combine these 2 flags into one setting:  `CreateOnStartAndDeleteOnStop` or `CreateAndDeleteSubscription` , something like that.

Reply:
Yes, makes kinda sense, I believe. `CreateOnStart_DeleteOnStop`, `Temporary`?

Reply:
We could introduce a new enumeration field with this new name, and deprecate the other two, as we haven't created a release with a warning of their deletion.

Reply:
`AutoCreateAndDelete` ?
I can work on this later this week.

Reply:
`AutoSetupSubscription`
`OwnsSubscription`
...

Reply:
Maybe `Automatic` will already be enough, as the enumeration already is called `TopicSubscription`?

Reply:
Ok.  
But I'm also thinking:  we have an enum now with only 2 options: 'None' and 'Automatic'.
Maybe this should be converted to a boolean flag ?

Reply:
It would take more effort in adding any additional option we're not thinking of, then to go from `bool` to enum again, right? Just thinking library develeopment, here. 😀

Reply:
That's true.  Just wondering what additional options will be added ....   But ok, let's stick with enum.

Reply:
Will resolve this as we stick with enum. 👍